main.tf

```diff
@@ -64,6 +65,29 @@ resource "aws_iam_role" "admin_role" {
 }
 EOF
 }
+resource "aws_iam_policy" "idp_guardrail" {
+  name        = "Idp_guardrail"
+  description = "Policy to restrict actions on providers"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Sid" : "AllowAdminAccess",
+        "Effect" : "Allow",
+        "Action" : "*",
+        "Resource" : "*"
+      },
+      {
+        "Sid" : "DenyPermBoundaryIAMIDPAlteration",
+        "Effect" : "Deny",
+        "Action" : "iam:*Provider"
+        "Resource" : [
+          "arn:aws:iam::${data.aws_caller_identity.aws_context.account_id}:role/*"
```

Comment:
We probably want to simply specify all resources with "*". I don't think the action will be on role resources.